src/components/modals/RedeemModal.tsx

```diff
@@ -131,35 +174,49 @@ export default function RedeemModal({
           </p>
         </div>
         <p>
-          <Trans>
-            Tokens can be redeemed for a portion of this project's ETH overflow,
-            according to the bonding curve rate of the current funding cycle.
-          </Trans>{' '}
-          <span style={{ fontWeight: 500, color: colors.text.warn }}>
-            {overflow?.eq(0) ? (
-              <Trans>
-                This project has no overflow, and you will receive no ETH for
-                burning tokens.
-              </Trans>
-            ) : (
-              <Trans>Tokens are burned when they are redeemed.</Trans>
-            )}
-          </span>
+          {overflow?.gt(0) ? (
+            <Trans>
+              Tokens can be redeemed for a portion of this project's ETH
+              overflow, according to the bonding curve rate of the current
+              funding cycle.{' '}
+              <span style={{ fontWeight: 500, color: colors.text.warn }}>
+                Tokens are burned when they are redeemed.
+              </span>
+            </Trans>
+          ) : (
+            <Trans>
+              <span style={{ fontWeight: 500, color: colors.text.warn }}>
+                <strong>This project has no overflow</strong>, so you will not
+                receive any ETH for burning tokens.
+              </span>
+            </Trans>
+          )}
         </p>
         <div>
-          <FormattedNumberInput
-            min={0}
-            step={0.001}
-            placeholder="0"
-            value={redeemAmount}
-            accessory={
-              <InputAccessoryButton
-                content={t`MAX`}
-                onClick={() => setRedeemAmount(fromWad(totalBalance))}
-              />
-            }
-            onChange={val => setRedeemAmount(val)}
-          />
+          <Form
+            form={form}
+            onKeyDown={e => {
+              if (e.key === 'Enter') redeem()
+            }}
+          >
+            <FormattedNumberInput
+              min={0}
+              step={0.001}
+              placeholder="0"
+              value={redeemAmount}
+              accessory={
+                <InputAccessoryButton
+                  content={t`MAX`}
+                  onClick={() => setRedeemAmount(fromWad(totalBalance))}
+                />
+              }
+              formItemProps={{
+                rules: [{ validator: validateRedeemAmount }],
+              }}
+              disabled={totalBalance?.eq(0)}
+              onChange={val => setRedeemAmount(val)}
+            />
+          </Form>
           <div style={{ fontWeight: 500, marginTop: 20 }}>
```

Comment:
maybe let's hide this for the "burn" state? seeing "You will receive 0" might add a bit of confusion to someone just trying to burn tokens.

Reply:
Agreed, it's unnecessary. Removed

Reply:
Yep agreed, it's unnecessary. Removed